Ensure Greek characters are transliterated

thomasleese opened the pull request:
At the moment there's an issue where Greek characters aren't been transliterated by ActiveSupport, and instead we end up with an empty string:

```rb
> "Στερεά Ελλάδα".parameterize
=> ""
```

This is an issue because this output is used the generate the slug of documents. After this change we get a more usable output:

```rb
> "Στερεά Ελλάδα".parameterize
=> "sterea-ellada"
```

I got the letters from https://gist.github.com/Achillefs/925267 and I'm not sure about the validity of them.

This has come up as part of the "Protected Food & Drink Names" finder where some titles are composed entirely of Greek letters and aren't able to be saved correctly because the slug ends up being an empty string.

thomasleese committed:
Ensure Greek characters are transliterated

At the moment there's an issue where Greek characters aren't been
transliterated by ActiveSupport, and instead we end up with an empty
string:

```
> "Στερεά Ελλάδα".parameterize
=> ""
```

This is an issue because this output is used the generate the slug of
documents. After this change we get a more usable output:

```
> "Στερεά Ελλάδα".parameterize
=> "sterea-ellada"
```

I got the letters from https://gist.github.com/Achillefs/925267 and I'm
not sure about the validity of them.

This has come up as part of the "Protected Food & Drink Names" finder
where some titles are composed entirely of Greek letters and aren't able
to be saved correctly because the slug ends up being an empty string.

ChrisBAshton reviewed (approved):
LGTM. Is it worth writing a test that `.parameterize` now does what you want it to? (You've already done it in the commit message, may as well put it in a unit test if possible, for extra confidence).

I hadn't come across Rails transliteration before. Nice one 👍

thomasleese committed:
Add a test for Greek letters

In 00abe551fd0c98a88c5447106e30545b234319f4 we added support for
transliterating Greek letters. This commit adds a test to ensure that
protected food & drink names made entirely of Greek letters get a
suitable base_path which was the original issue we were seeing.

richardTowers commented on config/locales/en.yml:
```diff
@@ -32,6 +32,108 @@
+        Χ: ch
+        ψ: ps
+        Ψ: ps
+        ω: w
+        ώ: w
+        Ώ: w
+        Ω: w
```
These ones are the only ones that stand out as possibly wrong to me. I think Omega is usually a long "o" sound (like in "toe"), so it should probably transliterated as "o", not "w".

richardTowers replied:
From my extensive duolingo greek 😬 

I'm sure we could find a native speaker if we wanted to be sure.

benjamineskola replied:
I also wondered about `ου` and `ευ` which if I remember correctly depend on the consonant that follows them — but I suppose it's close enough to be recognisable and it's pretty unambiguous.

ChrisBAshton replied:
Good spot! I guess it makes little difference in practice though, as it's only used in base_path creation.

thomasleese replied:
Thanks for this, I'll trust your judgement and open a PR to change the letters for omega!

thomasleese replied:
https://github.com/alphagov/specialist-publisher/pull/1771
